analyzer-core/src/main/java/ffb/analyzer/core/queue/PriorityQueueManager.java

```diff
@@ -0,0 +1,71 @@
+package ffb.analyzer.core.queue;
+
+import java.util.concurrent.PriorityBlockingQueue;
+
+public class PriorityQueueHandler{
+    private enum QueueState {
+        STOPPED,
+        RUNNING
+    }
+
+    private static final int MAX_CAPACITY = 1000;
+    private static final int STARTING_TASK_ID = 1;
+
+    private PriorityBlockingQueue<PriorityTask> queue;
+    private int nextAvailableId;
+    private QueueState state;
+
+    /**
+     * Adds entry to queue and updates tracking information within PriorityQueueHandler.
+     * @param entry Task object to be added to the queue.
+     * @return True if the entry was successfully added, otherwise False.
+     */
+    public boolean add(PriorityTask entry) {
+        incrementNextAvailableId();
+
+        return queue.add(entry);
+    }
+
+    public void incrementNextAvailableId() {
+        if (nextAvailableId == Integer.MAX_VALUE) {
```

Comment:
How about `nextAvailableId == Integer.MAX_VALUE ? Integer.STARTING_VALUE : nextAvailableId +1`?

Reply:
I actually just made it an AtomicInteger, so we could use the getAndIncrement method.